src/components/input/rating-picker/RatingPicker.tsx

```diff
@@ -0,0 +1,144 @@
+import classes from './RatingPicker.module.scss';
+
+import { Rating } from '../../../data/types/Rating';
+import { Heart, HeartHalf, IconProps } from '@phosphor-icons/react';
+import {
+    KeyboardEvent,
+    PointerEvent,
+    useEffect,
+    useRef,
+    useState,
+} from 'react';
+
+interface RatingPicker {
+    label: string;
+    rating: Rating;
+    onRatingChange: (rating: Rating) => void;
+}
+
+const RATING_ICON_PROPS: IconProps = {
+    size: 48,
+    color: 'var(--color-heart)',
+};
+
+const RatingPicker = ({ label, rating, onRatingChange }: RatingPicker) => {
+    const [pointerStartPosition, setPointerStartPosition] = useState<number>(0);
+    const [internalRating, setInternalRating] = useState<Rating>(rating);
+    const [previousIndex, setPreviousIndex] = useState<number>(0);
+
+    const ref = useRef<HTMLDivElement>(null);
```

Comment:
The name `ref` is too generic for a component that can have multiple refs. Please give it a more specific name.